src/Game/Scripting/Commands.cs

```diff
@@ -24,845 +24,1069 @@
-                throw new RunTimeError(null, "Usage: attack (serial)");
-
-            GameActions.Attack(args[0].AsSerial());
-
-            return true;
+            ScriptAction = 0,       // This command is related to a script logic unrelated to the game logic, such as aliases
+            ForceBypassQueue = 1,   // Force also means bypassing the queue
+            ComplexInterAction = 2, // This command is related to a complex interaction, like drag and drop something in game
+            SimpleInterAction = 4,  // This command is related to a simple interaction, like a click or dclick in an object
+            StateAction = 8,        // This command is related a game state, such as finding an object in the ground
+            PlayerAction = 16,      // This command is related to a player driven action, like moving or attacking
         }
+        public Attributes Attribute { get; }
 
-        private static bool ClearHands(string command, Argument[] args, bool quiet, bool force)
+        // Several flavors of constructors for quality of life
+        #region Constructors
+        public Command(string usage, Handler executionLogic, Handler waitLogic)
+            : this(usage, executionLogic, waitLogic, Attributes.ScriptAction)
         {
-            if (args.Length == 0 || !hands.Contains(args[0].AsString()))
-                throw new RunTimeError(null, "Usage: clearhands ('left'/'right'/'both')");
-
-            switch (args[0].AsString().ToLower())
-            {
-                case "left":
-                    GameActions.ClearEquipped(IO.ItemExt_PaperdollAppearance.Left);
-                    break;
-                case "right":
-                    GameActions.ClearEquipped(IO.ItemExt_PaperdollAppearance.Right);
-                    break;
-                // case "both":
-                default:
-                    GameActions.ClearEquipped(IO.ItemExt_PaperdollAppearance.Left);
-                    GameActions.ClearEquipped(IO.ItemExt_PaperdollAppearance.Right);
-                    break;
-            }
-
-            return true;
         }
-
-        private static bool ClickObject(string command, Argument[] args, bool quiet, bool force)
+        public Command(string usage, Handler executionLogic, Handler waitLogic, Attributes attribute)
         {
-            if (args.Length == 0)
-                throw new RunTimeError(null, "Usage: clickobject (serial)");
-
-            uint serial = args[0].AsSerial();
-
-            GameActions.SingleClick(serial);
+            Usage = usage;
+            ExecutionLogic = executionLogic;
+            WaitLogic = waitLogic;
+            Attribute = attribute;
+
+            // Make sure a queue exists for this command category
+            if (!Queues.ContainsKey(Attribute))
+                Queues.Add(Attribute, new Queue<CommandExecution>());
+
+            // Processing keywords and arguments in constructor to avoid logic on every command execution
```

Comment:
Are these two things needed? Or on error can it just print out the already-captured usage string? Parsing the usage string makes me a little nervous. It requires everything to be very consistent to not break.